File: problems/p852/p852.go
package p852

func peakIndexInMountainArray(arr []int) int {
	var left = 0
	var right = len(arr) - 1
	for left < right {
		var mid = (left + right) / 2
		if mid == left {
			return left
		}
		if arr[mid] > arr[mid-1] && arr[mid] < arr[mid+1] {
			left = mid
		} else if arr[mid] < arr[mid-1] && arr[mid] > arr[mid+1] {
			right = mid
		} else {
			return mid
		}
	}
	return left
}

File: problems/p852/p852.py
from typing import List


class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        left = 0
        right = len(arr)
        while left < right:
            mid = int((left + right)/2)
            if mid == left:
                return left
            if arr[mid+1] > arr[mid] > arr[mid-1]: 
                left = mid
            elif arr[mid+1] < arr[mid] < arr[mid-1]:
                right = mid
            else:
                return mid
        return left 